numpy/lib/function_base.py

```diff
@@ -140,6 +144,23 @@ def histogram(a, bins=10, range=None, normed=False, weights=None,
     second ``[2, 3)``.  The last bin, however, is ``[3, 4]``, which *includes*
     4.
 
+    The automated methods are well found in literature, and are inspired by the choices
+    R provides for histogram visualisation. Note that having the number of bins proportional
+    to $n^{1/3}$ is asymptotically optimal
+    fd (Freedman Diaconis) - depends on the IQR of the data and the cube root of datasize.
+        The Interquartile range (IQR) is very robust to noise. Can be too conservative
+        for small datasets however, but is quite good for larger ones.
+    scott - depends on the standard deviation of the data and the cube root of the datasize
+        The standard deviation includes information about data, but is less robust to noise that FD
+    rice - depends only on the cube root of the datasize
+        Can be overly loose as it does not take into account the variability of the data
+    sturges - depends on the base 2 log of the datasize
+        Assumes normality of data and is too conservative for larger datasets
+    auto - is the maximum of the sturges or fd estimators.
+        A compromise to get good automated bin choices. For small datasets the sturges value will be
+        chosen, while larger datasets will default to FD. Avoids the overly conservative values of
+        FD when small and Sturges when large. Switchover point is usually x.size~1000.
```

Comment:
For this block of text, the first three lines belong in the Notes section I think, and the rest as a reST definition list (http://sphinx-doc.org/rest.html#lists-and-quote-like-blocks) in the description of the ``bins`` parameter.